PROTOCOL_V6_DRAFT.md

```diff
@@ -0,0 +1,201 @@
+# EXPERIMENTAL Twirp v6 Wire Protocol
+
+This document is a draft of a new version ("v6") of the Twirp wire
+protocol over HTTP. This draft is EXPERIMENTAL and subject to change.
+v6 of Twirp is not released or considered stable; instead, this
+document lists planned changes which may be available in prerelease
+distributions.
+
+## Changes from v5
+
+### URL scheme
+
+In [v5](./PROTOCOL.md), URLs followed this format:
+
+**URL ::= Base-URL "/twirp/" [ Package "." ] Service "/" Method**
+
+In v6, the "/twirp/" prefix is removed:
+
+**URL ::= Base-URL "/" [ Package "." ] Service "/" Method**
```

Comment:
Please remove the first "/" and make it part of Base-URL. The less client logic, the better.

Reply:
I think we need the leading "/". If it's in the Base-URL, we need to say that the URL must not have an empty path. Otherwise, we'd allow a Base-URL of `https://example.com` which would lead to this:
`https://example.comtwirp.example.Haberdasher/MakeHat`

which seems clearly invalid.

Reply:
Wouldn't you use the `ResolveReference` method on `*url.URL` in order to support both cases?

Eg. https://play.golang.org/p/Zj6ehRCuZ_G

Reply:
@peterhellberg That might be a good way to make the API nice for Go users, but this question is on what the spec should be, which is a language-independent question. We're trying to work out what the BNF for the URL scheme should look like, which is probably a bit pedantic, but important to get right.